src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -252,8 +252,8 @@ private void insert(PreInitTrieNode node, String word, boolean negativeMatch) {
     }
 
     @Nullable
-    private TrieNode searchNode(byte[] bytes, int offset, int length) {
-        TrieNode node = root;
+    private TrieNode searchNode(TrieNode from, byte[] bytes, int offset, int length) {
```

Comment:
Would be nice if this had JavaDoc. Especially why offset and length are relevant in this context is not super clear from just the signature

Reply:
Added 👍